### PR TITLE
Added hardware detection for FreeBSD

### PR DESCRIPTION
**Pull request by frenchie**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/polling/os/unix.inc.php [hunk 1]
```diff
@@ -120,6 +120,25 @@
     if (!$features) {
         $features = 'GENERIC';
     }
+
+    # Try detect using the extended option (dmidecode)
+    $version_dmide = str_replace("\"", "", snmp_get($device, ".1.3.6.1.4.1.2021.7890.2.4.1.2.8.104.97.114.100.119.97.114.101.1", "-Oqv"));
```
Comment (murrant) [inline includes/polling/os/unix.inc.php]: Are all these str_replace() calls necessary?  The code above is legacy code, snmp_get() should have clean output.

Comment (frenchie) [reply includes/polling/os/unix.inc.php]: Ah good catch, I took the existing code for Linux distributions and adapted it for FreeBSD. Have just done a bit of testing and the code does work just fine without str_replace(). I will push a new commit that addresses my patch momentarily.

The rest of the file in question is similarly afflicted so I'll go through and clean that up and push a second commit for that also.

**Comment by murrant** [comment]
That's better, can you add test data for this, see auto link.

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-63-release-changelog-april-2020/11828/1

**Pull request merged**